python/test/test_polars.py

```diff
@@ -1127,3 +1127,22 @@ def test_enum_domain():
 
     # check that domain is as expected
     assert observed == expected
+
+
+def test_array_domain():
+    pl = pytest.importorskip("polars")
+
+    # this triggers construction of a lazyframe domain from the schema
+    context = dp.Context.compositor(
+        data=pl.LazyFrame(pl.Series("alpha", [["A", "B", "C"]] * 100, dtype=pl.Array(pl.String, 3))),
+        privacy_unit=dp.unit_of(contributions=1),
+        privacy_loss=dp.loss_of(epsilon=1.0, delta=1e-7),
+        split_evenly_over=1,
+    )
+    observed = context.accountant.input_domain
+    expected = dp.lazyframe_domain([
+        dp.series_domain("alpha", dp.option_domain(dp.array_domain(dp.atom_domain(T=str), 3))),
```

Comment:
To start the review, just copied this test out and ran it without any other changes, to understand what this PR gives us.

Not obvious why it should be wrapped in an option domain, but probably fine.

Reply:
This is for a dataframe column where each row is an array-- like the score vector emitted by the discrete quantile scoring transformation. Rows can be null instead of being an array. OpenDP makes as few assumptions about the data as possible, therefore it assumes that rows can be null instead of being an array.